File: pipeline/problems/3029a7bd-56ee-485e-b924-5895b9fdd5c6/Outputs/CodeContentFiles/Java/solution.java
import java.util.*;

public class Solution {
    public String compileDiagonalLedger(int m, int n, int[][] vaultGrid) {
        HashMap<Integer, ArrayList<Integer>> diagonals = new HashMap<>();

        for (int r = 0; r < m; r++) {
            if (r % 2 == 0) {
                for (int c = 0; c < n; c++) {
                    int key = r + c;
                    diagonals.putIfAbsent(key, new ArrayList<>());
                    diagonals.get(key).add(vaultGrid[r][c]);
                }
            } else {
                for (int c = n - 1; c >= 0; c--) {
                    int key = r + c;
                    diagonals.putIfAbsent(key, new ArrayList<>());
                    diagonals.get(key).add(vaultGrid[r][c]);
                }
            }
        }

        ArrayList<Integer> keys = new ArrayList<>(diagonals.keySet());
        Collections.sort(keys);

        StringBuilder result = new StringBuilder();
        for (int i = 0; i < keys.size(); i++) {
            int d = keys.get(i);
            ArrayList<Integer> group = diagonals.get(d);
            Collections.sort(group);

            int size = group.size();
            String val;
            if (size % 2 == 1) {
                int median = group.get(size / 2);
                val = String.format(Locale.US, "%.2f", (double) median);
            } else {
                long sum = 0;
                for (int x : group) sum += x;
                double avg = (double) sum / size;
                val = String.format(Locale.US, "%.2f", avg);
            }

            if (i > 0) result.append(" ");
            result.append(val);
        }

        return result.toString();
    }
}

File: pipeline/problems/3029a7bd-56ee-485e-b924-5895b9fdd5c6/Outputs/CodeContentFiles/NodeJS/solution.js
class Solution {
    static compileDiagonalLedger(m, n, vaultGrid) {
        const diagonals = new Map();

        for (let r = 0; r < m; r++) {
            if (r % 2 === 0) {
                for (let c = 0; c < n; c++) {
                    const key = r + c;
                    if (!diagonals.has(key)) diagonals.set(key, []);
                    diagonals.get(key).push(vaultGrid[r][c]);
                }
            } else {
                for (let c = n - 1; c >= 0; c--) {
                    const key = r + c;
                    if (!diagonals.has(key)) diagonals.set(key, []);
                    diagonals.get(key).push(vaultGrid[r][c]);
                }
            }
        }

        const keys = Array.from(diagonals.keys()).sort((a, b) => a - b);
        const result = [];

        for (let i = 0; i < keys.length; i++) {
            const group = diagonals.get(keys[i]).slice().sort((a, b) => a - b);
            const size = group.length;

            if (size % 2 === 1) {
                const median = group[Math.floor(size / 2)];
                result.push(median.toFixed(2));
            } else {
                let sum = 0;
                for (let j = 0; j < size; j++) sum += group[j];
                const avg = sum / size;
                result.push(avg.toFixed(2));
            }
        }

        return result.join(" ");
    }
}